angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.ts

```diff
@@ -216,6 +251,33 @@ export class MinesCollectionsAddEditComponent implements OnInit, OnDestroy {
     this.myForm.get('collectionRecords').markAsDirty();
   }
 
+  /**
+   * Add records to the collections array of associated records.
+   *
+   * @memberof MinesCollectionsAddEditComponent
+   */
+  addRecordsToCollection() {
```

Comment:
When you add a record to a collection, the state is preserved. So navigating away from the form and coming back will restore your selections. The only way to "clear" a selection set is to click cancel, which takes you to the `add\edit collection` screen. I think we should be able to clear the selection states without basically restarting the whole process.